File: Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/activities/settings/KnowledgeSettingsActivity.java
package com.salesforce.snapinssdkexample.activities.settings;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceFragmentCompat;
import com.salesforce.snapinssdkexample.R;

public class KnowledgeSettingsActivity extends AppCompatActivity {
    public static final String KEY_COMMUNITY_URL = "pref_kb_community_url";
    public static final String KEY_DATA_CATEGORY_GROUP = "pref_kb_data_category_group";
    public static final String KEY_ROOT_DATA_CATEGORY = "pref_kb_root_data_category";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getSupportFragmentManager().beginTransaction().replace(android.R.id.content, new SettingsFragment()).commit();
    }

    public static class SettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(@Nullable Bundle savedInstanceState, String rootKey) {
            addPreferencesFromResource(R.xml.prefs_knowledge_config);
        }
    }
}

File: Examples/SnapinsSDKExample/app/src/kotlinImplDebug/java/com/salesforce/snapinssdkexample/activities/settings/KnowledgeSettingsActivity.kt
/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

package com.salesforce.snapinssdkexample.activities.settings

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceFragmentCompat
import com.salesforce.snapinssdkexample.R

/**
 * Activity to display settings for knowledge.
 */
class KnowledgeSettingsActivity : AppCompatActivity() {
    companion object {
        val KEY_COMMUNITY_URL = "pref_kb_community_url"
        val KEY_DATA_CATEGORY_GROUP = "pref_kb_data_category_group"
        val KEY_ROOT_DATA_CATEGORY = "pref_kb_root_data_category"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportFragmentManager.beginTransaction().replace(android.R.id.content, SettingsFragment()).commit()
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            addPreferencesFromResource(R.xml.prefs_knowledge_config)
        }
    }
}
